content/api/commands/intercept.md

```diff
@@ -39,119 +39,29 @@ cy.intercept(url, routeMatcher, routeHandler)
-   * @default false
-   */
-  middleware?: boolean
-  /**
-   * Match on request path after the hostname, including query params.
-   */
-  path?: string | RegExp
-  /**
-   * Matches like 'path', but without query params.
-   */
-  pathname?: string | RegExp
-  /**
-   * Match based on requested port, or pass an array of ports
-   * to match against any in that array.
-   */
-  port?: number | number[]
-  /**
-   * Match on parsed querystring parameters.
-   */
-  query?: {
-    [key: string]: string | RegExp
-  }
-  /**
-   * Match against the full request URL.
-   */
-  url?: string | RegExp
-}
-```
-
-`routeMatcher` usage examples:
-
-```ts
-cy.intercept({
-  pathname: '/search',
-  query: {
-    q: 'some terms',
-  },
-}).as('searchForTerms')
-// this 'cy.wait' will only resolve once a request is made to '/search'
-// with the query paramater 'q=some+terms'
-cy.wait('@searchForTerms')
-
-cy.intercept(
-  {
-    // this RegExp matches any URL beginning with 'http://api.example.com/widgets'
-    url: /^http:\/\/api\.example\.com\/widgets/,
-    headers: {
-      'x-requested-with': 'exampleClient',
-    },
-  },
-  (req) => {
-    // only requests to URLs starting with 'http://api.example.com/widgets'
-    // having the header 'x-requested-with: exampleClient' will be received
-  }
-})
-
-// in this example, the supplied URL `/users` is merged with the RouteMatcher
-// passed as the second argument
-cy.intercept('/users', { middleware: true }, (req) => {
-  req.headers['authorization'] = `Bearer ${bearerToken}`
-})
-```
+| Option     | Default | Description                                                                        |
+|------------|---------|------------------------------------------------------------------------------------|
+| auth       | null    | `username` and `password` used in HTTP Basic authentication                        |
+| headers    | null    | HTTP request headers                                                               |
+| hostname   | null    | HTTP request hostname                                                              |
+| https      | null    | `true`: only secure (https://) requests `false`: only insecure (http://) requests  |
+| method     | "*"     | HTTP request method                                                                |
+| middleware | false   | Pass the request on to the next `RouteMatcher` after the request handler completes |
+| path       | null    | HTTP request path after the hostname, including query parameters                   |
+| pathname   | null    | Like `path`, but without query parameters                                          |
+| port       | null    | HTTP request port(s)                                                               |
+| query      | null    | Parsed query string parameters                                                     |
+| url        | null    | Full HTTP request URL                                                              |
```

Comment:
@amirrustam I updated this to be more concise and more consistent with the rest of the docs, but now it's really low-res compared to the typedef snippet. For example, we're not communicating where regex is another acceptable input. I thought about adding a `type` column and/or adding to the descriptions, but then it could be just as difficult to digest. Without these details, I would usually pull up the typedefs myself in vs code but I'm sure we can do better. I'm still working this out but thought I'd think aloud here.